### ISSUE TITLE
GRU behaviour varies for GPU and CPU

### ISSUE THREAD

**Issue by tejaswini** [pytorch/pytorch#4002]
When the number of layers in the GRU and the number of layers in the hidden size do not match, the GPU throws an error while the CPU does not.

The following snippet to illustrates the same:
```import torch
      rnn = torch.nn.GRU(input_size=5,hidden_size=4)
      x = torch.randn(3, 4, 5)
      hidden = torch.autograd.Variable(torch.randn(2, 4, 4))
      x = torch.autograd.Variable(x)
      if torch.cuda.is_available():
        rnn = rnn.cuda()
        x = x.cuda()
        hidden = hidden.cuda()
      rnn(x, hidden)
```

**Comment by zou3519** [comment]
I think this is a duplicate of https://github.com/pytorch/pytorch/issues/3851 (GRU uses the base RNN class). 

#3925 should fix this; the error message on the cpu will end up looking like:
![image](https://user-images.githubusercontent.com/5652049/33562010-2ed189b8-d8e2-11e7-8954-a71d1002dc82.png)

**Comment by zou3519** [comment]
Fixed by #3925
